### ISSUE TITLE
e-mail-confirmation setting to "no" not working

### ISSUE THREAD

**Issue by VOOM108** [catalyst/moodle-auth_enrolkey#37]
I am testing the plugin on a Moodle 3.5 Site. Everything is set up as described. When trying to sign up with an enrolment key, I get this error (with verbose debugging): 

![enrolmentkey-error01](https://user-images.githubusercontent.com/16562946/74357922-e6d50080-4dc0-11ea-84eb-f77dcea0cde2.png)

The new user is created, but he still needs to confirm his E-Mail, though that is set to "no" in the plugin settings. Also the user is not enrolled in the course with the resp. enrolment key. 

While there may also be a problem with our mailsettings, the plugin should not try to send any e-mail at all. Any idea, why the setting does not work?

**Comment by VOOM108** [comment]
Further testing: another site with Moodle 3.6.4 - everything works except the e-mail-setting: a confirmation mail is sent anyway and the account initially is not confirmed.

**Comment by brendanheywood** [comment]
There might be 2 things going on here which complicate each other, the first one is definitely an issue with sending email and would apply to all emails not just the signup ones.

In the second one can you provide some more detail, ie a screen shot of the settings page?

Also are you testing with the latest code from this repo and not the plugins directory?

**Comment by VOOM108** [comment]
I think we can ignore the first problem for now. I am using the version from this repo. These are the settings, but confirmation mails are sent anyway and it is necessary to click the confirmation link for full access: 

![aka-settings](https://user-images.githubusercontent.com/16562946/74784297-e8626500-52a7-11ea-8bc8-3f60eabd0da2.png)

**Comment by davidpesce** [comment]
Confirming the same thing occurs on Moodle 3.8.1+

To be clear, the value specified in the config settings for 'Require email confirmation' does not work.

**Comment by VOOM108** [comment]
I realized there is a difference between the settings: 

Set to "yes": the email is sent but the registration is not halted. Email-confirmation is still needed, but the course can be entered immediately.

Set to "no": the email is sent and the registration is halted. Only after confirming from the email the site can be entered and the user is enrolled in the course. However the course is not immediately entered.

**Comment by davidpesce** [comment]
Ahhh, I misunderstood. Would it be possible to have all three options?

- Direct access to course, but the user needs to click confirmation email for future logins
- Direct access to the course and the user is automatically confirmed (no confirmation email is sent nor required)
- No access to course until the user clicks confirmation email

**Comment by VOOM108** [comment]
Yes I did not get that, too, until I tried both. At least the wording for the setting is misleading. 

I would like to see those 3 options, too.... 

Also I wonder: now after registration the user gets an inbetween-page with an alert showing a link to the course. Is that extra step really necessary?

**Comment by brendanheywood** [comment]
@VOOM108 can you please retest using the latest master

**Comment by garotosopa** [comment]
I've just tested using the latest master and can confirm that this issue is now fixed - using the appropriate setting no longer requires email confirmation. Thank you.

Are there any plans to release a new version on Moodle.org? Is there anything we could help with?
